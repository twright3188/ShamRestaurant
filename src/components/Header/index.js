import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';

const styles = StyleSheet.create({
  headerStyles: {
    flex: 1,
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 24
  }
});

function Header(props) {
  const { children, style } = props;

  return (
    <View style={[styles.headerStyles, style]}>
      {children}
    </View>
  );
}

Header.propTypes = {
  children: PropTypes.node,
  style: PropTypes.shape({ root: PropTypes.string })
};

Header.defaultProps = {
  children: '',
  style: {}
};

export default Header;