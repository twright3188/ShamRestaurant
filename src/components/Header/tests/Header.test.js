import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../index';

it ('header renders correctly', () => {
    const tree = renderer.create(<Header>test</Header>).toJSON();
    expect(tree).toMatchSnapshot();
});