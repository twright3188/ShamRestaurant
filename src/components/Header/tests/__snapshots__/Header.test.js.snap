// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 24,
        "width": "100%",
      },
      Object {},
    ]
  }
>
  test
</View>
`;
